### ISSUE TITLE
CentOS Linux release 7.8.2003 (Core) installation error

### ISSUE THREAD

**Issue by sla-te** [agkozak/dotfiles#2]
after running ./install.sh which ran through fine, first start will drop:
```
agkozak-zsh-prompt: Using async method: subst-async

Downloading agkozak/zsh-z... (as agkozak/zsh-z...)

Cloning into '/root/.zinit/plugins/agkozak---zsh-z'...
▹▹▸▹▹ ███████████ OBJ: 100, PACK: 725/725, COMPR: 100%, REC: 100%, RES: 100%  
Unknown option: -C
usage: git [--version] [--help] [-c name=value]
           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
           [-p|--paginate|--no-pager] [--no-replace-objects] [--bare]
           [--git-dir=<path>] [--work-tree=<path>] [--namespace=<name>]
           <command> [<args>]
Compiling zsh-z.plugin.zsh.
_agkozak_precmd:47: scalar parameter LC_ALL created globally in function                                                               
_agkozak_precmd:47: scalar parameter GIT_OPTIONAL_LOCKS created globally in function

```
From then on all consecutive starts will drop:
```

Downloading zsh-users/zsh-history-substring-search... (as zsh-users/zsh-history-substring-search...)

Cloning into '/root/.zinit/plugins/zsh-users---zsh-history-substring-search'...
▄ ███████████ OBJ: 100, PACK: 391/391, REC: 100 RES: 100 
Compiling zsh-history-substring-search.plugin.zsh.
_agkozak_precmd:47: scalar parameter LC_ALL created globally in function                                                               
_agkozak_precmd:47: scalar parameter GIT_OPTIONAL_LOCKS created globally in function
```

**Comment by agkozak** [comment]
Thanks for letting me know.

It looks as if `zinit` has started to use the `-C` option of `git`, and your version of `git` is too old for that. We'll have to address that problem over in the [zdharma/zinit](https://github.com/zdharma/zinit) repository -- I'll be happy to do that, if you don't want to.

The warnings you're getting are associated with `setopt WARN_CREATE_GLOBAL`.

Could you give me your `git` and `zsh` versions?

    git --version
    zsh --version

Thanks! And thanks for helping me to improve my dotfiles!

**Comment by sla-te** [comment]
git version 1.8.3.1
zsh 5.0.2 (x86_64-redhat-linux-gnu)

Your dofiles are really good and very fast, I enjoy using them and appreciate the work you have put in them!

**Comment by agkozak** [comment]
Have you set `AGKOZAK_PROMPT_DEBUG=1` anywhere in your configuration? Just curious. Try entering

    echo $AGKOZAK_PROMPT_DEBUG

and see what it says.

I should have a solution for you; I just want to make sure I'm taking everything into consideration.

**Comment by agkozak** [comment]
And have you put something like `setopt WARN_CREATE_GLOBAL` anywhere in your dotfiles? If you type

    set -o

I suspect you'll see

    warncreateglobal         on

near the end of the list.

**Comment by sla-te** [comment]
- warncreateglobal is off
- echo $AGKOZAK_PROMPT_DEBUG is 1

note i have installed the default version of your dotfiles without any modifications and the above is only true if run from within zsh, from within bash $AGKOZAK_PROMPT_DEBUG would be empty and warncreateglobal would not exist in the list.

**Comment by agkozak** [comment]
Ah, yes -- I do set `AGKOZAK_PROMPT_DEBUG=1` in the `.zshrc`, don't I?

What's going on is the following. Since `AGKOZAK_PROMPT_DEBUG` is set to `1`, `WARN_CREATE_GLOBAL` is turned on in the function `_agkozak_precmd` in the agkozak ZSH prompt. Then, later in that function, we have the line

    : ${AGKOZAK_GIT_VERSION:=${"$(LC_ALL=C GIT_OPTIONAL_LOCKS=0 command git --version)"#git version }}

In any normal version of ZSH, that wouldn't set off `WARN_CREATE_GLOBAL` at all. But it does in ZSH 5.0.0-5.0.2. It's a bug in ZSH.

What I'll do to fix this problem is to release a new version of the agkozak ZSH prompt that checks the ZSH version before turning WARN_CREATE_GLOBAL on for that function.

The other problem -- the one involving `git -C` -- is entirely in `zinit`, so I'll open an issue in that repo and I'll let you know what gets decided. I imagine it's entirely a question of how old a version of Git the repo maintainer wants to support.

**Comment by agkozak** [comment]
Actually, ignore what I just wrote. I need to get a copy of CentOS up and running so that I can see if I can recreate the problem.

**Comment by agkozak** [comment]
OK. I've got the same version of CentOS running as you, and I see everything you're describing.

I have released [a new version of the agkozak ZSH prompt](https://github.com/agkozak/agkozak-zsh-prompt/releases/tag/v3.7.2) that avoids the warning you saw.

I agree that the behavior of Zinit is pretty strange on ZSH 5.0.2. I'm going to play around a bit and see how much of it I can fix myself, but we'll have to bring this issue up in the Zinit repo. I'll let you know what I figure out.

**Comment by agkozak** [comment]
I've got [a pull request in over at Zinit](https://github.com/zdharma/zinit/pull/341). I'll let you know when it works with ZSH v5.0.2 again. Thanks for you patience.

**Comment by sla-te** [comment]
Thanks man I really appreciate the effort!

**Comment by agkozak** [comment]
Just until we get Zinit working again for ZSH v5.0.7 and lower, I'm having my `.zshrc` use an alternate function to download and load my plugins and snippets. Give the new version a try and let me know if it works for you. I'll let you know when Zinit gets fixed.

**Comment by agkozak** [comment]
@chwba All right - sorry for the long wait. I'm trying to come up with a fix for `zinit` on `zsh` v5.0.2 again. I can fix the problem with your using an old version of Git, but there's something else that's wrong.

Could you tell me what you see when you type

    echo $signals

Thanks!

**Comment by agkozak** [comment]
I'm using my own plugin manager now ([`zcomet`](https://github.com/agkozak/zcomet)), so this shouldn't be a problem anymore.
